woocommerce/woocommerce-gutenberg-products-block
# Store Customization > Update the AI Prompts and structure of requests for Patterns, Products and Images

## What

On this PR, a series of changes were made to ensure the AI-generated content for products and patterns is more accurate and also introduce a significant performance boost by breaking down the multiple parallel requests to AI even further, ensuring it returns the expected results faster.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/11686

## Why


These changes are required to guarantee a better experience for the store owner when using the CYS flow for building their store.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

### Setup

- If you already have a pre-configured WordPress install, remove the pre-existing WooCommerce Blocks Plugin and upload the one from this zip file: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/13299024/woocommerce-gutenberg-products-block.zip)

Or alternatively, create a new WordPress install from scratch (make sure the install is a Jurassic Ninja install or equivalent. Otherwise you won't be able to connect with Jetpack to test this PR).

- Install and activate WooCommerce from the following zip file: [woocommerce.zip](https://github.com/woocommerce/woocommerce-blocks/files/13299011/woocommerce.zip)
- Install and activate WooCommerce Blocks from the following zip file: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/13299024/woocommerce-gutenberg-products-block.zip)

- Install and activate Jetpack.
- In your wp-admin, you should see the following section for Jetpack:

<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://user-images.githubusercontent.com/15730971/267093030-fc804759-f724-4fa2-829d-7bfd441962ec.png">

- Click on "Set up Jetpack" and make sure the connection is successful:

<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://user-images.githubusercontent.com/15730971/267093371-661ee4c5-a0ac-40fd-9e5e-835a0e4c1cb7.png">

- Install and activate Woo AI (the plugin is available [within the monorepo](https://github.com/woocommerce/woocommerce/tree/trunk/plugins))
- Install and activate WooCommerce Beta Tester (the plugin is available [within the monorepo](https://github.com/woocommerce/woocommerce/tree/trunk/plugins))
- Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag:

<img width="1233" alt="Screenshot 2023-10-22 at 10 32 24" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/816867c6-34e5-40a1-a87d-f4a7cda46429">

### Test the CYS experience

1. Head over to Home > Customize your Store:
 
<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/b5d34131-3edc-47ed-a8e4-8e2775ca1bbd">

2. Click on design with AI.
3. Type a description for your business.
4. Make sure the AI content is properly generated and images are assigned to patterns and products based on the description provided.
5. Repeat the same test multiple times with different business descriptions to ensure content is consistently updated every time.


## Screenshots or screencast




https://github.com/woocommerce/woocommerce-blocks/assets/15730971/c58bb673-a8e3-4883-9b80-0b9507cbde47




## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Customize your Store > Update the AI Prompts and structure of requests for Patterns, Products, and Images.
